### ISSUE TITLE
Revert polygon in Inspector in NavigationPolygonInstance doesn't delete this polygon

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#23644]
**Godot version:**
3.1 3e128a6

**OS/device including version:**
Linux Mint 19

**Issue description:**
When I try revert polygon from navigationpolygonInstance(2d), thenin inspector I see that this polygon was removed but in 2d viewport I still see it but I can't delete it and also show this error:
> ERROR: _navpoly_unlink: Condition ' !nm.linked ' is true.
   At: scene/2d/navigation2d.cpp:144.

**Steps to reproduce:**
I also try to revert collision shape and that works so removing polygon should do the same

https://streamable.com/2a1kw

**Comment by KoBeWi** [comment]
Works in 3.2, but throws
```
 servers/visual/visual_server_canvas.cpp:760 - Condition ' pointcount < 3 ' is true.
```

**Comment by KoBeWi** [comment]
Fixed in 3.2.4 beta3
Although a similar issue still exists: #44191
